New functions, bumped version number

ricardojmendez opened the pull request:
These commits:
- Bump the version number to 0.0.3-SNAPSHOT
- Add functions for browser/on-clicked, tabs/create and windows/get-current
- Add a storage namespace
- Update artifact references

And other small changes.

ricardojmendez committed:
Updating artifacts, bumping version

ricardojmendez committed:
Fixed typo

ricardojmendez committed:
New browser namespace, on-click function

ricardojmendez committed:
New windows namespace, get-current function

ricardojmendez committed:
New tabs/create function

ricardojmendez committed:
Improved khroma.tabs documentation, added tab-created-events

ricardojmendez committed:
browser/on-click and windows/get-current send values as Clojure map

The tab events we sending parameters keywordized and after passing them
through js->clj, doing the same for consistency.

ricardojmendez commented:
I have a few more commits, but for some reason github isn't detecting I've pushed them to my develop branch. I'll wait until these have been merged and create a different pull request.

ricardojmendez committed:
Bumping version on README

aav commented:
I will be back from a business trip on September 23, and will take care of all pull requests. Thank you for patience.

ricardojmendez committed:
New storage namespace

ricardojmendez committed:
Extracted tabs/tab-action-events to util/add-listener

The function is a good method of generally hooking up to Chrome events.
Changed browser/on-clicked to use it as well.

Renamed browser/on-click -> browser/on-clicked, to match the original
name.

TODO: Decide if the other addListener uses should remain as they are or
if they should use this general function. I'd say we err in the side of
consistency.

ricardojmendez committed:
New event: storage/on-changed

ricardojmendez committed:
Manual merge from master to solve conflicts from pull request #4

aav commented:
merged in, please check if anything was lost during conflict resolution

ricardojmendez commented:
Tested locally against the APIs I'm calling before pushing, it all worked as expected.   I'm not using 100%, though.

Of course being unable to do automated tests on a Chrome extension is a serious drawback. Perhaps it merits just writing an extension comprised solely of tests.

ricardojmendez commented:
Just noticed you meant you had to do a manual merge of your own.  There is a typo on line 24 of runtime.cljs on commit c7f26d0.

aav commented:
fixed!

aav commented:
regarding test extension it's a good idea, it also can work as a demo, how about todomvc? :)

ricardojmendez commented:
WIP extension works as expected.

Let's move the testing discussion to its own issue. I'll set some time aside to experiment with it.
